pkg/cmd/server/bootstrappolicy/infra_sa_policy.go

```diff
@@ -98,6 +98,10 @@ func init() {
 					Verbs:     sets.NewString("update"),
 					Resources: sets.NewString("builds"),
 				},
+				{
+					Verbs:     sets.NewString("create"),
+					Resources: sets.NewString("builds/docker", "builds/source", "builds/custom"),
+				},
```

Comment:
> so this is needed because the build controller updates builds? If so, it would make more sense to have this be an update permission

Not from a user perspective.  Granting the power to create docker builds is functionally equivalent to updating a build to be a docker build.  Why make me repeat myself?  These are virtual resources used for field level access control.

Now, why are build kinds mutable on a given build, that seems like a good question. :)